libnavigation-util/src/main/java/com/mapbox/navigation/utils/internal/ThreadController.kt

```diff
@@ -47,30 +47,37 @@ fun Exception.ifChannelException(action: () -> Unit) {
 
 data class JobControl(val job: Job, val scope: CoroutineScope)
 
-const val MAX_THREAD_COUNT = 2
+private const val MAX_THREAD_COUNT = 2
+private const val SINGLE_THREAD = 1
 
 object ThreadController {
     private val maxCoresUsed = Runtime.getRuntime().availableProcessors().coerceAtMost(
         MAX_THREAD_COUNT
     )
     val IODispatcher: CoroutineDispatcher =
         Executors.newFixedThreadPool(maxCoresUsed).asCoroutineDispatcher()
+    internal val navigatorDispatcher: CoroutineDispatcher =
```

Comment:
nit: throw a capital in there since it is a constant. appears more consistent with IODispatcher

```suggestion
    internal val NavigatorDispatcher: CoroutineDispatcher =
```